### PR TITLE
Perf opt

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: Xwt.XamMac/Xwt.Mac/ContextBackendHandler.cs [hunk 1]
```diff
@@ -38,15 +38,15 @@ class CGContextBackend {
 		public CGContext Context;
 		public CGSize Size;
 		public CGAffineTransform? InverseViewTransform;
-		public Stack<ContextStatus> StatusStack = new Stack<ContextStatus> ();
-		public ContextStatus CurrentStatus = new ContextStatus ();
+		public ContextStatus CurrentStatus;
```
Comment (slluis) [inline Xwt.XamMac/Xwt.Mac/ContextBackendHandler.cs]: CurrentStatus should never be null. It holds the currently active pattern.

Comment (Therzok) [inline Xwt.XamMac/Xwt.Mac/ContextBackendHandler.cs]: Yes, sorry about this.

File: Xwt.XamMac/Xwt.Mac/ContextBackendHandler.cs [hunk 2]
```diff
@@ -63,17 +63,19 @@ public override void Save (object backend)
...
+				Previous = ct.CurrentStatus,
+			};
 		}
 		
 		public override void Restore (object backend)
 		{
 			var ct = (CGContextBackend) backend;
 			ct.Context.RestoreState ();
-			ct.CurrentStatus = ct.StatusStack.Pop ();
+			if (ct.CurrentStatus != null) {
```
Comment (slluis) [inline Xwt.XamMac/Xwt.Mac/ContextBackendHandler.cs]: Now you should check `ct.CurrentStatus.Previous != null`

Comment (Therzok) [inline Xwt.XamMac/Xwt.Mac/ContextBackendHandler.cs]: *throws stuff at former self for not saving the file before git add -p*

**Pull request merged**